### ISSUE TITLE
Type aliased unnamed union in namespace causes invalid generated C++

### ISSUE THREAD

**Issue by Supreeeme** [google/autocxx#1261]
## C++ header
```c++
#pragma once

namespace test {

typedef union {
    int a;
} Union;

}
```

## Rust code
```rust
use autocxx::prelude::*;

include_cpp! {
    #include "test.h"
    generate!("test::Union")
}

fn main() {
}
```
## Expected Behavior
Successful compilation or a more descriptive error.

## Actual Behavior
The following error is emitted:
```
cargo:warning=In file included from /tmp/cxx_test/target/debug/build/cxx_test-734c6d667e055e4a/out/autocxx-build-dir/cxx/gen0.cxx:2:
  cargo:warning=/tmp/cxx_test/target/debug/build/cxx_test-734c6d667e055e4a/out/autocxx-build-dir/include/autocxxgen_ffi.h: In function ‘void Union_synthetic_destructor_0x322cc83aeed3c411_autocxx_wrapper_0x322cc83aeed3c411(test::Union*)’:
  cargo:warning=/tmp/cxx_test/target/debug/build/cxx_test-734c6d667e055e4a/out/autocxx-build-dir/include/autocxxgen_ffi.h:44:128: error: expected class-name before ‘(’ token
  cargo:warning=   44 | inline void Union_synthetic_destructor_0x322cc83aeed3c411_autocxx_wrapper_0x322cc83aeed3c411(test::Union* arg0)  { arg0->~Union(); }
  cargo:warning=      |                                                                                                                                ^
  exit status: 1
```
This could be fixed by adding a `using test::Union;` right before the destructor call.
Also, I can't use `generate_pod!` on the type, which I thought would be possible, because autocxx says this:
```
Type test::Union could not be POD because its dependent type __BindgenUnionField isn't known" }
```

**Comment by adetaylor** [comment]
Thanks - do you think you could [raise a PR containing a minimal test case](https://github.com/google/autocxx/blob/main/book/src/contributing.md#reporting-bugs)?

**Comment by adetaylor** [comment]
I'm about to be travelling for a couple of weeks so it will be a while before I can resolve the final problems in that PR and merge it.

**Comment by adetaylor** [comment]
I can't actually find a way to call the destructor of an unnamed type (at least not one which works for both the Linux and OS X compilers I usually use).

[godbolt example](https://godbolt.org/z/sh5Y5G78h)

So that's annoying.

**Comment by Supreeeme** [comment]
I don't have an OS X compiler to test on but if you bring it into scope it works: https://godbolt.org/z/GG99aKffW

**Comment by adetaylor** [comment]
Unfortunately, that doesn't work if we have two structs called the same name, in different namespaces. I suppose we could track this case and report an error, but I'd much rather give a fully qualified name for the destructor.

If there's really no alternative, I suppose I could make a function like this for every struct:

```cpp
void destruct_a_A(a::A* ptr) {
  using namespace a;
  a->A::~A();
}
```

but that would be pretty tedious.

**Comment by adetaylor** [comment]
Fixed by #1261, hopefully. It was a bit of a journey. Thanks for the report.
